File: 0601-0700/0606-Construct String from Binary Tree/0606-Construct String from Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    string tree2str(TreeNode* t) {
        string result;
        if (t == nullptr) {
            return "";
        }
        
        result += to_string(t->val);
        if (t->right) {
            result += '(' + tree2str(t->left) + ')';
            result += '(' + tree2str(t->right) + ')';
        }
        else if (t->left) {
            result += '(' + tree2str(t->left) + ')';
        }
        
        return result;
    }
};

File: 0601-0700/0606-Construct String from Binary Tree/0606-Construct String from Binary Tree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def tree2str(self, t: TreeNode) -> str:
        if t is None:
            return ''
        left = '({})'.format(self.tree2str(t.left)) if t.left or t.right else ''
        right = '({})'.format(self.tree2str(t.right)) if t.right else ''
        return '{}{}{}'.format(str(t.val), left, right)
